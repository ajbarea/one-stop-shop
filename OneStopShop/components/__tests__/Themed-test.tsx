import * as React from 'react';
import renderer from 'react-test-renderer';
import { Text, View } from '../Themed';
import * as useColorSchemeModule from '../useColorScheme';

jest.mock('../useColorScheme');

describe('Themed components', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('Text component renders correctly', () => {
    const tree = renderer.create(<Text>Snapshot test for Text!</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('View component renders correctly', () => {
    const tree = renderer.create(<View><Text>Snapshot test for View!</Text></View>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Text component uses lightColor prop when provided', () => {
    const tree = renderer.create(<Text lightColor="red">Test with lightColor</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('View component uses darkColor prop when provided', () => {
    const tree = renderer.create(<View darkColor="blue"><Text>Test with darkColor</Text></View>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Text component uses default theme when useColorScheme returns null', () => {
    (useColorSchemeModule.useColorScheme as jest.Mock).mockReturnValue(null);
    const tree = renderer.create(<Text>Test with default theme</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Text component uses light theme when useColorScheme returns light', () => {
    (useColorSchemeModule.useColorScheme as jest.Mock).mockReturnValue('light');
    const tree = renderer.create(<Text>Test with light theme</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});