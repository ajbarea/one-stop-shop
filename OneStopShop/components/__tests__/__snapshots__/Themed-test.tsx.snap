// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themed components Text component renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  Snapshot test for Text!
</Text>
`;

exports[`Themed components Text component uses default theme when useColorScheme returns null 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  Test with default theme
</Text>
`;

exports[`Themed components Text component uses light theme when useColorScheme returns light 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  Test with light theme
</Text>
`;

exports[`Themed components Text component uses lightColor prop when provided 1`] = `
<Text
  style={
    [
      {
        "color": "red",
      },
      undefined,
    ]
  }
>
  Test with lightColor
</Text>
`;

exports[`Themed components View component renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    Snapshot test for View!
  </Text>
</View>
`;

exports[`Themed components View component uses darkColor prop when provided 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    Test with darkColor
  </Text>
</View>
`;
